test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,317 @@
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertIn("field required", response.text.lower())
+        self.assertEqual(response.status_code, 422)
+        logger.info("=== Test Case 8 Completed ===\n")
+
+    def test_missing_chat_id(self):
+        """Test case 7: Missing chat_id"""
+        logger.info("=== Starting Test Case 7: Missing chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertIn("field required", response.text.lower())
+        self.assertEqual(response.status_code, 422)
+        logger.info("=== Test Case 7 Completed ===\n")
+
+    def test_invalid_page_number(self):
+        """Test case 9: Invalid Page Number"""
+        logger.info("=== Starting Test Case 9: Invalid Page Number ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 0,
+            "page_size": 10
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("no chats found for this customer and chat id", response.text.lower())
+        logger.info("=== Test Case 9 Completed ===\n")
+
+    def test_invalid_page_size(self):
+        """Test case 10: Invalid Page Size"""
+        logger.info("=== Starting Test Case 10: Invalid Page Size ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 1,
+            "page_size": 0
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+
+        self.assertEqual(response.status_code, 404)
+        logger.info("=== Test Case 10 Completed ===\n")
```

Comment:
Can we also verify the output text here

Reply:
Done sir, I've added the assertion to verify the response text.